src/logger/connectivityTask.c

```diff
@@ -373,6 +373,15 @@ void connectivityTask(void *params)
                 pr_info_str_msg(connParams->connectionName, ": timeout");
                 should_reconnect = true;
             }
+
+            /*
+             * If we didn't rx or tx a message, sleep for a bit so as not
+             * starve the other tasks.
+             */
+            if (pdFALSE == res && !msgReceived) {
```

Comment:
The current code will always block if there is no message to send, but if there is a message received and needing a reply, it will wait for the above send to timeout, then process the received message.  This change fixes that by only sleeping when we both don't send anything and don't receive anything.  

What this is trying to solve is to not wait unnecessarily to process messages received by the BT stack.  In effect it is a minor optimization.  I tested it yesterday and BT responded a bit faster, but was still more sluggish than I would like.